wicker/core/datasets.py

```diff
@@ -152,33 +225,42 @@ def get_folder_size(bucket, prefix):
         return get_folder_size(bucket, key)
 
     def _get_dataset_size(self):
-        """Gets total size of the dataset in bits
+        """Gets total size of the dataset in bits.
 
         Returns:
             int: total dataset size in bits
         """
+
         # intialize with size of parquet dir
+        print("Parsing parquet and arrow dir for size.")
         par_dir_bytes = self._get_parquet_dir_size()
 
         # need to know which columns are heavy pntr columns we go to for
-        # byte adding
+        # byte adding, parse which are heavy pointers based off metadata
+        # baked into schema
         schema = self.schema()
         heavy_pointer_cols = []
         for col_name in schema.get_all_column_names():
-            if schema.get_column(col_name).is_heavy_pointer:
-                heavy_pointer_cols.append(col_name)
+            column = schema.get_column(col_name)
+            if column is not None:
+                if column.is_heavy_pointer:
+                    heavy_pointer_cols.append(col_name)
 
-        # Each individual row only knows which column file it goes to, so we have to
-        # neccesarily parse all rows :( to get the column files. This should be cached
-        # as metadata but that would require re-curating the datasets.
+        # create arrow table for parsing
         print("Creating arrow table")
         arrow_table = self.arrow_table()
 
         buckets_keys = set()
-        worker = ShuffleWorker(storage=self._storage)
+
+        # ignore typing to avoid changing the typing of Shuffle Worker yet
+        # ToDo: Change typing of ShuffleWorker and let it take local data storage
+        worker = ShuffleWorker(storage=self._storage)  # type: ignore
```

Comment:
@pickles-bread-and-butter if you look at the code, the `ShuffleWorker` here never really did anything. It is only used to access the `S3PathFactory`. Wouldn't it be better to get rid of it and just create the `S3PathFactory` directly (or perhaps there is one defined already in `_self`)?